python/xgrammar/matcher.py

```diff
@@ -400,3 +400,122 @@ def _debug_print_internal_state(self) -> str:
+    def batched_fill_next_token_bitmask(
+        matchers: List["GrammarMatcher"],
+        bitmask: ArrayLike,
+        index: int = 0,
+        max_threads: int = 16,
+        debug_print: bool = False,
+    ) -> List[bool]:
+        """Fill the next token bitmask for multiple matchers.
+
+        Parameters
+        ----------
+        matchers : List[GrammarMatcher]
+            The list of matchers to fill the bitmask for.
+
+        bitmask : ArrayLike
+            The bitmask to fill.
+
+        index : int, default: 0
+
+        max_threads : int, default: 16
+            The maximum number of threads to use for filling the bitmask.
+                    index : int, default: 0
+            The batch id of the bitmask.
+
+        debug_print : bool, default: False
+            Whether to print information about generated bitmask. Helpful for debugging.
+                    Returns
+
+        Returns
+        -------
+        need_apply : List[bool]
+            Whether the bitmask need to be applied (not all-true). An optimization: if False,
+            this means the bitmask is already all-true, so no need to apply it.
+
+        Raises
+        ------
+        RuntimeError
+            If the bitmask is invalid (not on CPU, not int32, shape mismatch).
+
+            If the recursion depth is exceeded.
+        """
+        matcher_handles = [matcher._handle for matcher in matchers]
+
+        return _core.GrammarMatcher.batched_fill_next_token_bitmask(
+            matcher_handles, bitmask, index, max_threads, debug_print
+        )
+
+    @staticmethod
+    def batched_accept_token(
+        matchers: List["GrammarMatcher"], tokens: List[int], debug_print: bool = False
+    ) -> List[bool]:
+        """Accept a batch of tokens for multiple matchers.
+
+        Parameters
+        ----------
+        matchers : List[GrammarMatcher]
+            The list of matchers to accept tokens for.
+
+        tokens : List[int]
+            The list of tokens to accept.
+
+        debug_print : bool, default: False
+            Whether to print information about generated bitmask. Helpful for debugging.
+
+        Returns
+        -------
+        accepted : List[bool]
+            A list of booleans indicating whether each token was accepted by its corresponding matcher.
+
+        Raises
+        ------
+        RuntimeError
+            If the sizes of matchers and tokens do not match.
+        """
+        if len(matchers) != len(tokens):
+            raise RuntimeError(
+                "The sizes of matchers and tokens do not match. "
+                + f"Got {len(matchers)} matchers and {len(tokens)} tokens."
+            )
+
+        matcher_handles = [matcher._handle for matcher in matchers]
+        return _core.GrammarMatcher.batched_accept_token(matcher_handles, tokens, debug_print)
+
+    @staticmethod
+    def batched_accept_string(
+        matchers: List["GrammarMatcher"], strings: List[str], debug_print: bool = False
```

Comment:
```suggestion
        matchers: List["GrammarMatcher"], strings: List[Union[byte, str]], debug_print: bool = False
```
batch_accept_string

If hard to implement, consider Union[List[byte], List[str]]